benchmark/checks/check_mem_usage.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2020 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""Memory usage check."""
+import time
+from threading import Thread
+
+import click
+
+from aea.protocols.base import Message
+from aea.protocols.default.message import DefaultMessage
+from aea.skills.base import Handler
+
+from benchmark.checks.utils import (  # noqa: I100
+    SyncedGeneratorConnection,
+    get_mem_usage_in_mb,
+    make_agent,
+    make_envelope,
+    make_skill,
+    multi_run,
+    print_results,
+    wait_for_condition,
+)
+
+
+class TestHandler(Handler):
+    """Dummy handler to handle messages."""
+
+    SUPPORTED_PROTOCOL = DefaultMessage.protocol_id
+
+    def setup(self) -> None:
+        """Noop setup."""
+
+    def teardown(self) -> None:
+        """Noop teardown."""
+
+    def handle(self, message: Message) -> None:
+        """Handle incoming message."""
+        self.context.outbox.put(make_envelope(message.to, message.sender))
+
+
+def run(duration, runtime_mode):
+    """Check memory usage."""
```

Comment:
For the memory usage; can we add the reactive and proactive here? So this would also run the behaviour at the same time.